clang/include/clang/CConv/ProgramInfo.h

```diff
@@ -56,10 +56,12 @@ class ProgramInfo {
   // For each pointer type in the declaration of D, add a variable to the 
   // constraint system for that pointer type.
   void addVariable(clang::DeclaratorDecl *D, clang::ASTContext *astContext);
+  void addCompoundLiteral(CompoundLiteralExpr *CLE, ASTContext *astContext);
```

Comment:
I find it surprising that you are adding a `CompoundLIteralExpr` to `ProgramInfo`. I think of this class as containing the "environment" of a type system, i.e., the type constraints, and a map from variables to their types (including constraint variables). The solution to the constraint variables means rewriting the types. 

From this perspective, then: Why are compound literals being added to `ProgramInfo` ? Is it because you need to (always) rewrite a type nearby? That doesn't seem sufficient justification to me, since we also have to insert casts sometimes, e.g., at function calls, and we don't store function call expressions in ProgramInfo.

Reply:
I see why you are doing this now. Related point: Should we be keeping track of casts too, in case we need to rewrite those?